### ISSUE TITLE
[Craft 5] If I miss a required field while saving a old entry, the Entry structure breaks 

### ISSUE THREAD

**Issue by AleksanderKampus1** [spicywebau/craft-neo#929]
### Bug Description

This bug is related to this bug: https://github.com/spicywebau/craft-neo/issues/907 

While working on a old entry in Craft CMS, let's say I forget inputting a required field (such as component title). When I try to save the entry, I get a validation error. All's good, but now when I fix the validation error and try to save the entry again, something weird happens. The Neo structure breaks. 

It's important to note that it only only happens to entries that where saved before updating to Neo 5.2.0. Meaning that when I make a totally new entry everything works. That's why I didn't catch it with the original bug report. 

### Steps to reproduce

1. Set autoSaveDrafts to false in CMS config 
2. Choose a old entry
3. Don't put in a required field
4. Save it. Craft will give a validation error
5. Fix that validation error and save it again
6. Neo blocks will get messed up 

https://github.com/user-attachments/assets/6f32f7cb-5cae-4e5a-8dab-059bc325dc97

Here is also a video if I create a new entry: 

https://github.com/user-attachments/assets/6041871d-9e12-461c-b97e-7b9160adbe68

### Expected behaviour

Neo blocks should maintain the original structure  

### Neo version

5.2.0

### Craft CMS version

5.3.4

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by ttempleton** [comment]
I'm unable to reproduce this - but if you run the Craft [`resave/entries`](https://craftcms.com/docs/5.x/reference/cli.html#resave-entries) command, does that resolve it?

**Comment by AleksanderKampus1** [comment]
We tried running the resave for a single entry, but unfortunately, it didn't help with the issue. Is there anything we can provide to help you debug it more effectively?

We updated Neo from version 5.0.4 to 5.2.2 and Craft from version 5.2.8 to 5.3.5. Could that cause any problems?

**Comment by AleksanderKampus1** [comment]
Hey @ttempleton! I have been playing around with it more and I have found a way to break the structure in also new entries. Maybe this is a new bug or is somehow related to this. 

Steps to reproduce:  

1. Set autoSaveDrafts to false in CMS config
2. Create a new entry with an component that has a required field and don't fill the required field
3. Save it. Craft will give a validation error
4. Fix that validation error and save it again
5. Everything should be fine
6. Now delete the component that had the required field
7. Add the component back but don't fill the required field 
8. Save and you should get a validation error
9. Fill the require field 
10. Save and the whole structure gets messed up 
11. Refresh
12. Now you don't see the structure messed up again but you also don't see the component anymore
13. In addition now the entry works

**Comment by ttempleton** [comment]
Does updating to 5.2.4 and running `php craft gc/run` resolve the issue? (in case this has the same underlying cause as #927)

**Comment by AleksanderKampus1** [comment]
Unfortunately this didn't fix it :(. Tried also 5.2.5

**Comment by kubanedo** [comment]
I am encountering the same problem :(.

**Comment by AleksanderKampus1** [comment]
Hey @kubanedo! If you don't mind, can you send Neo team ([plugins@spicyweb.com.au](mailto:plugins@spicyweb.com.au)) your db backup and composer files, so they can look more into it. I have problems sending mine, so I can't unfortunately do it :(

**Comment by ttempleton** [comment]
Yes - could you please send those files in if possible @kubanedo.

**Comment by AleksanderKampus1** [comment]
Hey, @krftcz, I see that you added an like, are you experiencing the same issue. If yes, can you maybe send the database and composer files? So they can look into it?

@ttempleton also did you test out these steps I mentioned earlier: https://github.com/spicywebau/craft-neo/issues/929#issuecomment-2321352842

**Comment by ttempleton** [comment]
Thanks @AleksanderKampus1 - I've tried that out with a parent and child block where the child block has the required field, and the parent block has disappeared. I'll assume this is the same bug.

**Comment by kubanedo** [comment]
Hi @AleksanderKampus1 ,

sorry for the late reply, the requested DB backup was sent to the email address. Hopefully, it will help solve the issue :).

Thanks

**Comment by kubanedo** [comment]
Hi @AleksanderKampus1 ,

do you can think of any workaround we could you in case we would need to deploy our project on production next week and the issue wasn't solved yet?

Thanks a lot

**Comment by AleksanderKampus1** [comment]
Hey @kubanedo ! 
The only workaround I know of is to use lower Craft-Neo version. We are using 5.0.4 and this doesn't have that issue. 
Hope this helps 👍 . 

Also did you send your composer files with the DB backup?

**Comment by kubanedo** [comment]
Oh, great, will try.

Of course I didn't :D. I've sent it now.

Thanks

**Comment by kubanedo** [comment]
Hi @AleksanderKampus1, any update?
Thanks

**Comment by AleksanderKampus1** [comment]
Hey @kubanedo ! I am just using `craft-neo` the same as you are so I don't have any extra information on it. But from this issue thread what I understand is that @ttempleton was able to reproduce it (I don't know if using your db dump). So I am guessing the Neo team is now working on it.

**Comment by ttempleton** [comment]
Thanks for sending the files @kubanedo. I'm unable to reproduce this issue with your DB using the original steps to reproduce for the issue, but I'm going to try to fix the issue from https://github.com/spicywebau/craft-neo/issues/929#issuecomment-2321352842, in case it's the same issue, before I ask for any more info.

**Comment by AleksanderKampus1** [comment]
Hey! @ttempleton so I tried out some latest versions and it seems that in 5.2.11 you did something that made things better. Now the structure doesn't get totally jumbled up but what still happens is that the component that had validation error gets set as the most top component. Like lets say I have this kinda structure: 
1. Container
2. Container 2
3. Component with validation

After I fix the validation error the structure gets to set like this: 

1. Component with validation
2. Container
3. Container 2

Hope it helps fixing the problem. I can send a screen recording if it is a bit confusing what I explained.
